src/net/types.ts

```diff
@@ -194,12 +197,26 @@ export interface ITransportPipelines {
 
   image: {
     loader: (x : ISegmentLoaderArguments) =>
-      ILoaderObservable<Uint8Array|ArrayBuffer>;
-    parser: (x : ISegmentParserArguments<Uint8Array|ArrayBuffer>) =>
+      ILoaderObservable<ArrayBuffer|Uint8Array>;
+    parser: (x : ISegmentParserArguments<ArrayBuffer|Uint8Array>) =>
       ImageParserObservable;
   };
 }
 
+interface IParsedKeySystem {
+  systemId : string;
+  privateData : Uint8Array;
+}
+
+export interface IParserOptions {
```

Comment:
The problem here is that some are HSS-specific and other DASH-specific. We should work to try to have the same options for both